source/lambda/documentbulkprocessor/lambda_function.py

```diff
@@ -12,16 +12,177 @@
+    return True
+
+
+def processDocument(record):
+    
+    print(str(record['s3']['object']['key']))
+
+    ingestionBucketName = record['s3']['bucket']['name']
+    ingestionDocumentKey = record['s3']['object']['key']
+    ingestionDocumentFilename = record['s3']['object']['key'].split('/')[1]
+
+    s3Client = boto3.client('s3', region_name=os.environ['AWS_REGION'])
+    
+    # get document from bulk bucket
+    ret = downloadDocument(ingestionBucketName,
+                           ingestionDocumentKey,
+                           ingestionDocumentFilename,
+                           s3Client)
+
+    # error trying to dowload document, exit
+    if ret == False:
+        print("processDocument: failed to download locally document:" + ingestionDocumentKey)
+        return
+    
+    # attempt to get an optional document Kendra policy json file
+    policyFilename = ingestionDocumentFilename + ".metadata.json"
+    policyKey = "kendraPolicyDrop/" + policyFilename
+
+    s3helper = S3Helper()
+    policyData = None
+
+    # try to fetch amd load the policy
+    try:
+        
+        policyData = s3helper.readFromS3(ingestionBucketName,
+                                       policyKey,
+                                       os.environ['AWS_REGION'])
+
+
+    # the normal case of a file not provided is handled.  If any other error
+    # occur the indexing will proceed without the membership tags in the policy file
+    except ClientError as e:
+        policyData = None
+        # NoSuchKey is the expected exception when no policy provided
+        if e.response['Error']['Code'] == 'NoSuchKey':
+            print("no kendra policy file found, skipping")
+        else:
+            print("ClientError exception from s3helper.readFromS3() for policy file: " + str(e))
 
-        print("event: {}".format(event))
+    # an error that should be investigated
     except Exception as e:
-        print("Failed to process records. Exception: {}".format(e))
+        policyData = None
+        print("unspecified exception from s3helper.readFromS3() for policy file: " + str(e))
+                                    
+    # generate UUID for document
+    documentId = generateDocumentID(os.environ['OUTPUT_BUCKET'], s3Client)
+
+    # upload document in document bucket
+    destinationDocumentKey = "public/" + documentId + "/" + ingestionDocumentFilename
+
+    ret = uploadDocument(os.environ['OUTPUT_BUCKET'],
+                         destinationDocumentKey,
+                         ingestionDocumentFilename,
+                         s3Client)
+
+    # in case of error uploading document, we exit
+    if ret == False:
+        print("failed to upload document into output bucket: " + destinationDocumentKey)
+        return
+
+    # if optional Kendra policy was present, upload it to document folder
+    # alongside document
+    if policyData != None:
+        print("copying over to document folder the kendra policy file")
```

Comment:
Could please add relevant ids in all the log statements? 
For example adding document Id here could provide a lot of value in debugging.